src/GraphQL/Validation/ValidationOptions.cs

```diff
@@ -11,7 +11,9 @@ public class ValidationOptions
     {
         public ISchema Schema { get; set; } = null!;
 
-        public Document Document { get; set; } = null!;
+        public GraphQLDocument Document { get; set; } = null!;
+
+        public string? OriginalQuery { get; set; }
```

Comment:
I added OriginalQuery here and in other places because `GraphQLDocument` from parser does not hold this property. I think it is even better.

Reply:
I think we may want to revert this design change.  Since the parser returns ROMs everywhere, there is no reason not to also hold a reference to the original query, since each ROM does anyway.  And it makes for cleaner design here in GraphQL.NET.

Reply:
Hold a reference to what exactly? Parser/Lexer works with `ROM` as input, not with `string`. It's important. `ROM` may be backed by memory buffer.

> since each ROM does anyway

No _anyway_. I introduced all these `ROM`-stuff with a long-term goal in the future to be able to work with memory buffers.

Reply:
Closing as duplicate of another conversation. I propose to add `ROM Source {get;}` property into `GraphQLDocument` and revert some of changes discussed above.